src/components/SubHeader/EvergreenTaskSubHeader.tsx

```diff
@@ -0,0 +1,106 @@
+import { useQuery } from "@apollo/client";
+import { InlineCode } from "@leafygreen-ui/typography";
+import { usePreferencesAnalytics } from "analytics";
+import Breadcrumbs from "components/Breadcrumbs";
+import Icon from "components/Icon";
+import TaskStatusBadge from "components/TaskStatusBadge";
+import { LogTypes } from "constants/enums";
+import { getEvergreenTaskURL } from "constants/externalURLTemplates";
+import {
+  LogkeeperTaskQuery,
+  LogkeeperTaskQueryVariables,
+  TaskQuery,
+  TaskQueryVariables,
+} from "gql/generated/types";
+import { GET_LOGKEEPER_TASK, GET_TASK } from "gql/queries";
+import { shortenGithash } from "utils/string";
+
+interface Props {
+  buildID: string;
+  execution: number;
+  logType?: LogTypes;
+  taskID: string;
+  testID?: string;
+}
+
+export const EvergreenTaskSubHeader: React.FC<Props> = ({
+  buildID,
+  execution,
+  logType,
+  taskID,
+  testID,
+}) => {
+  const { sendEvent } = usePreferencesAnalytics();
+  const { data, loading } = useQuery<TaskQuery, TaskQueryVariables>(GET_TASK, {
+    variables: { taskId: taskID, execution },
+    skip: logType === LogTypes.RESMOKE_LOGS,
+  });
+  const { data: logkeeperData, loading: logkeeperLoading } = useQuery<
+    LogkeeperTaskQuery,
+    LogkeeperTaskQueryVariables
+  >(GET_LOGKEEPER_TASK, {
+    variables: { buildId: buildID },
+    skip: logType !== LogTypes.RESMOKE_LOGS || !buildID,
+  });
+
+  const { task } = data ?? {};
+  const { logkeeperBuildMetadata } = logkeeperData ?? {};
+
+  if (loading || logkeeperLoading || (!task && !logkeeperBuildMetadata)) {
+    return <Icon glyph="EvergreenLogo" size={24} />;
+  }
+
+  // @ts-expect-error - We verify above that either task or logkeeperBuildMetadata exists, so logkeeperBuildMetadata will not be undefined as TypeScript suspects
+  const loadedTask = task ?? logkeeperBuildMetadata.task;
+  const { displayName, patchNumber, status } = loadedTask;
+  const { isPatch, projectIdentifier, message, revision } =
+    loadedTask.versionMetadata ?? {};
+
+  const currentTest = logkeeperBuildMetadata?.tests?.find(
+    (test) => test.id === testID
+  );
+
+  const breadcrumbs = [
+    {
+      text: projectIdentifier,
+      "data-cy": "project-breadcrumb",
+    },
+    {
+      tooltipText: message,
+      "data-cy": "version-breadcrumb",
+      text: isPatch ? (
+        `Patch ${patchNumber}`
+      ) : (
+        <InlineCode>{shortenGithash(revision)}</InlineCode>
+      ),
+    },
+    {
+      href: getEvergreenTaskURL(taskID, execution),
+      text: (
+        <>
+          {displayName} <TaskStatusBadge status={status} />
+        </>
+      ),
+      "data-cy": "task-breadcrumb",
+      onClick: () => {
+        sendEvent({ name: "Opened Task Link" });
+      },
+    },
+    ...(testID
+      ? [
+          {
+            "data-cy": "test-breadcrumb",
+            tooltipText: currentTest?.name,
```

Comment:
I went with the tooltip approach because test names quickly got suuuuper long 😬 open to ideas but this seemed like a quick and straightforward approach!

Reply:
I feel like being able to quickly see the test name might be valuable when viewing different resmoke test logs in multiple tabs. One option could be to allow the test name specifically to have a higher character cut off than 30?

<img width="1210" alt="Screenshot 2023-04-07 at 10 38 33 AM" src="https://user-images.githubusercontent.com/47064971/230628640-f0c6e67a-0ce1-43c7-aaf0-74e63ee36cff.png">

Reply:
done